src/main/java/fi/helsinki/cs/tmc/core/communication/ExerciseDownloader.java

```diff
@@ -19,6 +14,13 @@
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
```

Comment:
Nit: imports are not sorted correctly and make the checkstyle fail..
> Import statements are divided into the following groups, in this order, with each group separated by a single blank line:

>All static imports in a single group
1) fi.helsinki.cs.tmc imports (only if this source file is in the fi.helsinki.cs.tmc package space)
2) Third-party imports, one group per top-level package, in ASCII sort order
3) for example: android, com, junit, org, sun
4) java imports
5) javax imports